kong/clustering/services/sync/rpc.lua

```diff
@@ -243,71 +294,24 @@ local function do_sync()
   -- { type = ..., entity = { ... }, version = "1", ws_id = ..., }
   for _, delta in ipairs(deltas) do
     local delta_version = delta.version
-    local delta_type = delta.type
     local delta_entity = delta.entity
-    local ev
+    local delta_type = delta.type
 
     -- delta should have ws_id to generate the correct lmdb key
     -- if entity is workspaceable
     -- set the correct workspace for item
     opts.workspace = delta.ws_id
 
-    if delta_entity ~= nil and delta_entity ~= ngx_null then
-      -- upsert the entity
-      -- does the entity already exists?
-      local old_entity, err = db[delta_type]:select(delta_entity)
-      if err then
-        return nil, err
-      end
-
-      -- If we will wipe lmdb, we don't need to delete it from lmdb.
-      if old_entity and not wipe then
-        local res, err = delete_entity_for_txn(t, delta_type, old_entity, opts)
-        if not res then
-          return nil, err
-        end
-      end
-
-      local res, err = insert_entity_for_txn(t, delta_type, delta_entity, opts)
-      if not res then
-        return nil, err
-      end
-
-      ngx_log(ngx_DEBUG,
-              "[kong.sync.v2] update entity",
-              ", version: ", delta_version,
-              ", type: ", delta_type)
-
-      -- wipe the whole lmdb, should not have events
-      if not wipe then
-        ev = { delta_type, old_entity and "update" or "create", delta_entity, old_entity, }
-      end
-
-    else
-      -- delete the entity, opts for getting correct lmdb key
-      local old_entity, err = db[delta_type]:select(delta.pk, opts) -- composite key
-      if err then
-        return nil, err
-      end
-
-      -- If we will wipe lmdb, we don't need to delete it from lmdb.
-      if old_entity and not wipe then
-        local res, err = delete_entity_for_txn(t, delta_type, old_entity, opts)
-        if not res then
-          return nil, err
-        end
-      end
-
-      ngx_log(ngx_DEBUG,
-              "[kong.sync.v2] delete entity",
-              ", version: ", delta_version,
-              ", type: ", delta_type)
-
-      -- wipe the whole lmdb, should not have events
-      if not wipe then
-        ev = { delta_type, "delete", old_entity, }
-      end
-    end -- if delta_entity ~= nil and delta_entity ~= ngx_null
+    local is_upsert = delta_entity ~= nil and delta_entity ~= ngx_null
+    local operation_name = is_upsert and "upsert" or "delete"
+    local operation = is_upsert and lmdb_upsert or lmdb_delete
+
+    ngx_log(ngx_DEBUG,
```

Comment:
This log should below the `operation()`.

Reply:
It will then be a copy&paste which is an antipattern

Reply:
No, log should only after  `operation()` successes.

Reply:
This way we cannot know what operation is failing